rlclientlib/federation/vw_local_joiner.cc

```diff
@@ -0,0 +1,222 @@
+#include "federation/vw_local_joiner.h"
+
+#include "constants.h"
+#include "err_constants.h"
+#include "logger/message_type.h"
+#include "logger/preamble.h"
+#include "trace_logger.h"
+#include "vw/config/options_cli.h"
+#include "vw/core/cache.h"
+#include "vw/core/parse_primitives.h"
+
+namespace reinforcement_learning
+{
+vw_local_joiner::vw_local_joiner(const utility::configuration& config, i_trace* trace_logger, api_status* status)
+    : _trace_logger(trace_logger)
+{
+  if (config.get_int(name::PROTOCOL_VERSION, 999) != 2)
+  {
+    TRACE_ERROR(_trace_logger, "Protocol version 2 is required");
+    // TODO handle error
+    // Can't return error code in constructor
+  }
+
+  std::string cmd_str = config.get(name::MODEL_VW_INITIAL_COMMAND_LINE,
+      "--cb_explore_adf --driver_output_off --epsilon 0.2 --power_t 0 -l 0.001 --cb_type mtr -q :: "
+      "--preserve-performance-counters");
+  auto cmd_list = VW::split_command_line(cmd_str);
+  auto options = VW::make_unique<VW::config::options_cli>(cmd_list);
+  _joiner_workspace = VW::initialize_experimental(std::move(options));
```

Comment:
Does it make sense for the joiner to need a `Workspace` instance? I think we should probably avoid this